core/domain/draft_upgrade_services.py

```diff
@@ -0,0 +1,131 @@
+    commits_list = (
+        exp_models.ExplorationCommitLogEntryModel.get_multi(
+            exp_id, exp_versions))
+    upgrade_times = 0
+    while current_draft_version + upgrade_times < to_exp_version:
+        commit = commits_list[upgrade_times]
+        if (
+                len(commit.commit_cmds) == 1 and
+                commit.commit_cmds[0]['cmd'] ==
+                exp_domain.CMD_MIGRATE_STATES_SCHEMA_TO_LATEST_VERSION):
+
+
+            conversion_fn_name = '_convert_states_v%s_dict_to_v%s_dict' % (
+                commit.commit_cmds[0]['from_version'],
+                commit.commit_cmds[0]['to_version'])
+            if not hasattr(DraftUpgradeUtil, conversion_fn_name):
+                logging.warning('%s is not implemented' % conversion_fn_name)
+                return
+            conversion_fn = getattr(DraftUpgradeUtil, conversion_fn_name)
+            draft_change_list = conversion_fn(draft_change_list)
+        else:
+            return
+        upgrade_times += 1
+    return draft_change_list
+
+
+class DraftUpgradeUtil(object):
+    """Wrapper class that contains util functions to upgrade drafts."""
+
+    @classmethod
+    def _convert_states_v28_dict_to_v29_dict(cls, draft_change_list):
+        """Converts draft change list from state version 28 to 29. State
+        version 29 adds solicit_answer_details boolean variable to the
+        state, for which there should be no changes to drafts.
+
+        Args:
+            draft_change_list: list(ExplorationChange). The list of
+            ExplorationChange domain objects to upgrade.
+
+        Returns:
+            list(ExplorationChange). The converted draft_change_list.
+        """
+        return draft_change_list
+
+    @classmethod
+    def _convert_states_v27_dict_to_v28_dict(cls, draft_change_list):
+        """Converts draft change list from state version 27 to 28. State
+        version 28 replaces content_ids_to_audio_translations with
+        recorded_voiceovers.
+
+        Args:
+            draft_change_list: list(ExplorationChange). The list of
+            ExplorationChange domain objects to upgrade.
+
+        Returns:
+            list(ExplorationChange). The converted draft_change_list.
+        """
+        for i, change in enumerate(draft_change_list):
+            if (change.cmd == exp_domain.CMD_EDIT_STATE_PROPERTY and
+                    change.property_name ==
+                    exp_domain.STATE_PROPERTY_CONTENT_IDS_TO_AUDIO_TRANSLATIONS_DEPRECATED):  # pylint: disable=line-too-long
+                draft_change_list[i] = exp_domain.ExplorationChange({
+                    'cmd': exp_domain.CMD_EDIT_STATE_PROPERTY,
+                    'property_name': (
+                        exp_domain.STATE_PROPERTY_RECORDED_VOICEOVERS),
+                    'state_name': change.state_name,
+                    'new_value': {
+                        'voiceovers_mapping': (
+                            change.new_value)
```

Comment:
Nit: not sure you need to wrap the line, you can probably put this at the end of the previous line and remove the parens.

Reply:
Done.